llvm/lib/Transforms/InstCombine/InstCombineSelect.cpp

```diff
@@ -3112,6 +3112,26 @@ Instruction *InstCombinerImpl::foldSelectOfBools(SelectInst &SI) {
     return SelectInst::Create(FalseVal, One, AndV);
   }
 
+  // select (~b & a), a, b -> or a, b
+  // only for scalar types
+  if (match(CondVal, m_c_And(m_Not(m_Specific(FalseVal)), m_Specific(TrueVal))) &&
```

Comment:
This pattern is similar to, but more specific than, the one on line 3109. You should probably move this transform above that one.

Reply:
Correction: Both your patterns are more specific than the ones between lines 3089 and 3109. Move your code above those.